toxcore/events/events_alloc.c

```diff
@@ -78,3 +80,23 @@ bool tox_events_add(Tox_Events *events, const Tox_Event *event)
 
     return true;
 }
+
+bool clone_byte_array(uint8_t **output, uint32_t *output_size, const uint8_t *input, uint32_t input_size)
+{
+    if (*output != nullptr) {
+        free(*output);
+        *output = nullptr;
+        *output_size = 0;
+    }
+
+    uint8_t *input_copy = (uint8_t *)malloc(input_size);
+
+    if (input_copy == nullptr) {
+        return false;
+    }
+
+    memcpy(input_copy, input, input_size);
```

Comment:
`input` is specified as nullable, but there is no check for it being `nullptr`, and passing `nullptr` to `memcpy` is UB.

Reply:
I want a linter for this :P. Yes, this check is necessary (it actually happens in the file transfer callbacks, which is why it's nullable in the first place).